code/obj/item/implant.dm

```diff
@@ -316,24 +316,30 @@ THROWING DARTS
 		if(inafterlife(src.owner))
 			return
 		DEBUG_MESSAGE("[src] calling to report crit")
-		health_alert()
+		SPAWN(rand(15, 25) SECONDS)
+			health_alert()
 		..()
 
 	on_death()
 		if(inafterlife(src.owner))
 			return
 		DEBUG_MESSAGE("[src] calling to report death")
-		death_alert()
+		SPAWN(rand(15, 25) SECONDS)
+			death_alert()
 		..()
 
 	proc/health_alert()
 		if (!src.owner)
 			return
+		if (!src)
+			return
 		src.send_message("HEALTH ALERT: [src.owner] in [get_area(src)]: [src.sensehealth()]", MGA_MEDCRIT, "HEALTH-MAILBOT")
 
 	proc/death_alert()
 		if (!src.owner)
 			return
+		if (!src)
+			return
```

Comment:
???

Reply:
Wasn't sure if the line of code before it prevented a runtime if the implant was deleted, so I put those there. If it does, I'll remove em.

Reply:
You'd want to check `QDELETED(src)` for that, but the prior check should in theory catch that, as `owner` gets nulled out in disposing